### PR TITLE
Remove duplicate code from controllers [WIP]

### LINKED ISSUE #4224: Clean up identical code in codeclimate
There are several codeclimate "smells" as detailed in #3706. This task will focus on cleaning up duplicate code.

To do this, we will need to remove duplicate code at:
- [ ] app/controllers/tag_controller.rb line 221 & line 155
- [ ] doc/app/js/searcher.js 115 & 128 &  89
- [ ] app/assets/javascripts/application/i18n/translations.js 2 &  3
- [ ] app/assets/javascripts/question.js 25 &  32
- [ ] app/controllers/admin_controller.rb 35 &  21
- [ ] app/assets/javascripts/jsdiff.js 128 & 139
- [ ] doc/app/js/navigation.js 46 & 55
- [ ] app/controllers/wiki_controller.rb 413 & 426
- [ ] app/assets/javascripts/dragdrop.js 129 & 89
- [ ] app/api/srch/search.rb 264 & 296
- [ ] app/models/node.rb 666 & 695
- [ ] app/api/srch/search.rb 119 & 204
- [ ] app/api/srch/search.rb 147 & 327
- [ ] app/assets/javascripts/like.js 44 & 28
- [ ] app/models/concerns/node_shared.rb 354 & 69 & & 12
- [ ] app/assets/javascripts/jsdiff.js 152 & 144
- [ ] app/models/concerns/node_shared.rb 182 & 215
- [ ] doc/app/js/searcher.js 161 & 139

Keep in mind that though the code in these areas are similar, you can't simply delete the repeated code. Consider creating a function if a variable is different, and always test before checking off any of these issues. 

This covers all the duplicate code, as far as I am able to find within code climate. Please let me know if there's anything else to change.

### PR DESCRIPTION
**Pull request by ValentinaTironi**
Fixes #4224 

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [ ] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] PR is descriptively titled 📑
* [ ] screenshots/GIFs are attached 📎 in case of UI updation
* [ ] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

**Comment by ValentinaTironi** [comment]
OK, I have to adapt the test

**Comment by gauravano** [comment]
Looking great!

**Comment by ValentinaTironi** [comment]
I can follow the codeclimate's sugestions, because the code doesn't the same thing, What . I have to do? @gauravano

**Comment by gauravano** [comment]
I will take a look. Thanks!

File: app/controllers/admin_controller.rb [hunk 1]
```diff
@@ -389,4 +331,43 @@ def smtp_test
     s.close
   end
 
+  def moderator_or_admin_user
```
Comment (jywarren) [inline app/controllers/admin_controller.rb]: Should these maybe all go under a `private` section so they are only call-able from within this file?

File: app/controllers/tag_controller.rb [hunk 2]
```diff
@@ -218,22 +203,7 @@ def show_for_author
...
-          json.last['path'] = 'https://' + request.host
-            .to_s + node.path
-          json.last['preview'] = node.body_preview(500)
-          json.last['image'] = node.main_image.path(:large) if node.main_image
-          json.last['tags'] = Node.find(node.id).tags.collect(&:name) if node.tags
-        end
-        render json: json
-      end
-    end
+    respond_with_nodes(nodes)
```
Comment (jywarren) [inline app/controllers/tag_controller.rb]: Maybe we could rename this something like "respond_with_formats" instead?

File: app/controllers/wiki_controller.rb [hunk 3]
```diff
@@ -410,19 +410,6 @@ def methods
       .references(:node_revision)
       .order('node_revisions.timestamp DESC')
     # deprecating the following in favor of javascript implementation in /app/assets/javascripts/methods.js
-    if params[:topic]
```
Comment (jywarren) [inline app/controllers/wiki_controller.rb]: Whoa was this just completely duplicated????

**Comment by jywarren** [comment]
no problem, and thanks again!

**Comment by SidharthBansal** [comment]
As the person is inactive for more than a month, I am closing the PR. In case you want to push changes please feel free to open a new PR OR reopen this PR and add additional changes to it.
Thanks for contributing on Public Lab
cc @jywarren @gauravano

**Comment by gauravano** [comment]
Actually, I like the work done in this PR, the clean-up is awesome. I will complete this PR. Thanks @SidharthBansal!

**Comment by SidharthBansal** [comment]
Any update on this @gauravano ?

**Comment by gauravano** [comment]
I guess I will close it for now and will complete it when get some free time. Thanks @SidharthBansal!

**Pull request closed without merging**